Library/Homebrew/extend/os/linux/cleanup.rb

```diff
@@ -2,8 +2,10 @@
 # frozen_string_literal: true
 
 module Homebrew
-  class Cleanup
-    undef use_system_ruby?
+  module CleanupLinux
```

Comment:
```suggestion
  module Linux::Cleanup
```
or
```suggestion
  module Cleanup::Linux
```
or
```suggestion
  module Cleanup::LinuxExtensions
```
or maybe even
```suggestion
  module OS::Linux::Cleanup
```

Reply:
`OS::Linux::Cleanup` most closely matches the path, let's go with that one

Reply:
Welp, one issue with introducing `OS` into the name space is that breaks resolution of e.g. `OS.linux?` under `Homebrew`, including in outside repos, e.g. https://github.com/Homebrew/homebrew-test-bot/blob/cb38bc3/lib/test_bot.rb#L63

We could alias `::OS` to point to `Homebrew::OS`, but wanted to do a sanity check before pursuing this route any further.

Reply:
I couldn't get a simple alias to work, but this does:
```
module Homebrew
  module OS
    class << self
      extend Forwardable
      def_delegators '::OS', :mac?, :linux?, :kernel_version, :kerneal_name, :unsupported_configuration?
    end
  end
end
```
We'd presumably then deprecate top-level `::OS`, given the desire to move everything under the `Homebrew` namespace.

Reply:
> We'd presumably then deprecate top-level `::OS`, given the desire to move everything under the `Homebrew` namespace.

Would that deprecate `OS.linux?` and `OS.mac?`? If so, then that would break _a lot_ of code, and not just in Homebrew.

Reply:
> `OS::Linux::Cleanup` most closely matches the path, let's go with that one

@dduugg sounds good, thanks!

---

I think having a top-level `OS` even just as an alias to `Homebrew::OS` would be good. As @carlocab mentioned, it's been an incredibly heavily used public API for a long time to the point that I'd probably rather never deprecate/disable/remove it but keep around the alias indefinitely (like we do with `MacOS` to `OS::Mac`).

If the alias isn't possible: feel free to leave this as-is and avoid the Homebrew namespace (or do so just for the `extend` modules)

Reply:
I've moved all the OS extensions under the top-level `OS` in the latest diff, which avoids having to do full-path resolution all over the place. I was ~forced to change `OS::MacOS` to `OS::Mac`, because `::MacOS` is already defined as a [constant](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/os/mac.rb#L16) under `OS`, which would create more namespace resolution errors.